MetadataExtractor/Formats/QuickTime/QuickTimeMetadataReader.cs

```diff
@@ -176,6 +178,22 @@ void Handler(AtomCallbackArgs a)
                         QuickTimeReader.ProcessAtoms(stream, MoovHandler, a.BytesLeft);
                         break;
                     }
+                    case "uuid":
+                    {
+                        var XMP = new byte[] { 0xbe, 0x7a, 0xcf, 0xcb, 0x97, 0xa9, 0x42, 0xe8, 0x9c, 0x71, 0x99, 0x94, 0x91, 0xe3, 0xaf, 0xac };
+                        if (a.BytesLeft >= XMP.Length)
+                        {
+                            var uuid = a.Reader.GetBytes(XMP.Length);
+                            if (XMP.RegionEquals(0, XMP.Length, uuid))
+                            {
+                                var xmpBytes = a.Reader.GetNullTerminatedBytes((int)a.BytesLeft);
+                                var xmpDirectory = new XmpReader().Extract(xmpBytes);
+                                xmpDirectory.Parent = directories.OfType<ExifIfd0Directory>().FirstOrDefault();
```

Comment:
I'm still not sure this is correct. The parent here should be the metadata container that wrapped this inner collection. In this case we are in a QuickTime container which does not have a directory, so I would just leave the parent null and remove this line.

```suggestion
```

Reply:
The idea here was to mimic the directory structure seen in CR2. I was actually considering using `Directories.Insert()`  to put `xmpDirectory` in front of its siblings.

Reply:
Can you help me understand the use case for needing similar structure?

Reply:
There is not a strict *need*, just that some applications that process CR2 might find it handy.